File: app/src/main/kotlin/xyz/digicard/app/android/NewAdapter.java
package xyz.digicard.app.android;

import android.support.v4.content.res.ResourcesCompat;
import android.support.v4.graphics.drawable.DrawableCompat;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.List;

import xyz.digicard.app.android.models.User;

/**
 * Created by Chris on 1/27/2018.
 */

public class NewAdapter extends RecyclerView.Adapter<NewAdapter.ViewHolder> {

    private List<User> data;

    public static class ViewHolder extends RecyclerView.ViewHolder{
        private TextView firstName, lastName, email;
        private ImageView imgView;
        public ViewHolder(View v){
            super(v);
            this.firstName = (TextView)v.findViewById(R.id.contact_first_name);
            this.lastName = (TextView)v.findViewById(R.id.contact_last_name);
            this.imgView = (ImageView)v.findViewById(R.id.profile_image);
            this.email = (TextView)v.findViewById(R.id.email);
        }
        public void bind(User u){
            firstName.setText(u.getFirstName());
            lastName.setText(u.getLastName());
            imgView.setImageDrawable(ResourcesCompat.getDrawable(imgView.getContext().getResources(), R.drawable.user_photo_fallback, imgView.getContext().getTheme()));
            email.setText(u.getEmail());
        }
    }
    public NewAdapter(List<User> dataset){
        data = dataset;
    }
    @Override
    public NewAdapter.ViewHolder onCreateViewHolder(ViewGroup parent,
                                                   int viewType) {
        // create a new view
        View v = (View) LayoutInflater.from(parent.getContext())
                .inflate(R.layout.user_profile_contact_container, parent, false);
        // set the view's size, margins, paddings and layout parameters
        ViewHolder vh = new ViewHolder(v);
        return vh;
    }

    // Replace the contents of a view (invoked by the layout manager)
    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        // - get element from your dataset at this position
        // - replace the contents of the view with that element
        holder.bind(data.get(position));

    }

    // Return the size of your dataset (invoked by the layout manager)
    @Override
    public int getItemCount() {
        return data.size();
    }
}

File: app/src/main/kotlin/xyz/digicard/app/android/NewAdapter.kt
package xyz.digicard.app.android

import android.support.v4.content.res.ResourcesCompat
import android.support.v4.graphics.drawable.DrawableCompat
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import xyz.digicard.app.android.NewAdapter.ViewHolder

import xyz.digicard.app.android.models.User

class NewAdapter(private val data: List<User>) : RecyclerView.Adapter<NewAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NewAdapter.ViewHolder {
        return ViewHolder(LayoutInflater
                .from(parent.context)
                .inflate(R.layout.user_profile_contact_container, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(data[position])
    }

    override fun getItemCount(): Int {
        return data.size
    }


    class ViewHolder(v: View) : RecyclerView.ViewHolder(v) {

        private val firstName: TextView
        private val lastName: TextView
        private val email: TextView
        private val imgView: ImageView
        private lateinit var user: User

        init {
            v.setOnClickListener { _ ->
                ConnectedUserActivity.startActivity(v.context, user.id)
            }

            this.firstName = v.findViewById<View>(R.id.contact_first_name) as TextView
            this.lastName = v.findViewById<View>(R.id.contact_last_name) as TextView
            this.imgView = v.findViewById<View>(R.id.profile_image) as ImageView
            this.email = v.findViewById<View>(R.id.email) as TextView
        }

        fun bind(u: User) {
            user = u
            firstName.text = u.firstName
            lastName.text = u.lastName
            email.text = u.email

            imgView.setImageDrawable(ResourcesCompat.getDrawable(
                    imgView.context.resources,
                    R.drawable.user_photo_fallback,
                    imgView.context.theme
            ))
        }
    }
}


